react/features/remote-control/components/RemoteControlAuthorizationDialog.js

```diff
@@ -0,0 +1,165 @@
+         *
+         * @private
+         */
+        _displayName: React.PropTypes.string,
+
+        /**
+         * Used to show hide the dialog on cancel.
+         */
+        dispatch: React.PropTypes.func,
+
+        /**
+         * The ID of the participant who is requesting authorization for remote
+         * desktop control session.
+         *
+         * @public
+         */
+        participantId: React.PropTypes.string,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func
+    }
+
+    /**
+     * Initializes a new RemoteControlAuthorizationDialog instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this._onCancel = this._onCancel.bind(this);
+        this._onSubmit = this._onSubmit.bind(this);
+    }
+
+    /**
+     * Renders additional message text for the dialog.
+     *
+     * @private
+     * @returns {ReactElement}
+     */
+    _getAdditionalMessage() {
+        // FIXME: Once we have this information in redux we should
+        // start getting it from there.
+        if (APP.conference.isSharingScreen) {
+            return null;
+        }
+
+        return (
+            <div>
+                <br />
+                { this.props.t('dialog.remoteControlShareScreenWarning') }
+            </div>
+        );
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        return (
+            <Dialog
+                cancelTitleKey = { 'dialog.Cancel' }
+                okTitleKey = { 'dialog.Allow' }
+                onCancel = { this._onCancel }
+                onSubmit = { this._onSubmit }
+                titleKey = 'dialog.remoteControlTitle'
+                width = 'small'>
+                {
+                    this.props.t('dialog.remoteControlRequestMessage',
+                    { user: this.props._displayName })
+                }
+                {
+                    this._getAdditionalMessage()
+                }
+            </Dialog>
+        );
+    }
+
+    /**
+     * Cleans existing preview tracks and signal to closeDeviceSelectionDialog.
+     *
+     * @private
+     * @returns {boolean} Returns false to prevent closure until cleanup is
+     * complete.
+     */
+    _onCancel() {
+        // FIXME: This should be action one day.
+        APP.remoteControl.receiver.deny(this.props.participantId);
+
+        return true;
+    }
+
+    /**
+     * Identify changes to the preferred input/output devices and perform
+     * necessary cleanup and requests to use those devices. Closes the modal
+     * after cleanup and device change requests complete.
+     *
+     * @private
+     * @returns {boolean} Returns false to prevent closure until cleanup is
+     * complete.
+     */
+    _onSubmit() {
```

Comment:
Haha! I forgot to change the comment!

You are correct for returning true and this was my first implementation but I hit the following issue:
1. The authorization dialog opens and the user haven't started the desktop sharing
2. The user clicks Allow
3. The desktop picker dialog is supposed to open but the redux action for opening it is ignored because the authorization dialog is still displayed. 

To workaround this I had to hide the authorization dialog before dispatching the action for the desktop picker.

I'll fix the comments only.